server/src/main/java/org/opensearch/index/store/Store.java

```diff
@@ -1190,6 +1202,45 @@ public RecoveryDiff recoveryDiff(MetadataSnapshot recoveryTargetSnapshot) {
             return recoveryDiff;
         }
 
+        /**
+         * Returns a diff between the two snapshots that can be used for getting list of files to copy over to a replica for segment replication. The given snapshot is treated as the
+         * target and this snapshot as the source.
+         */
+        public RecoveryDiff getFilesRecoveryDiff(MetadataSnapshot recoveryTargetSnapshot) {
```

Comment:
We'll need unit tests on this method.

Reply:
Yes, although the reason is different - not because of checksum - it marks all identical files as different even when they're not if there are any missing files that exist between the snapshots.